### ISSUE TITLE
No way to 'unmaximise' window

### ISSUE THREAD

**Issue by ukch** [wilfm/GnomeExtensionMaximusTwo#16]
When maximising a window using this plugin, the window icons disappear, which means that for most applications it is now impossible to unmaximise or otherwise resize the window.

One option (I don't know if this is possible) would be to add window icons to the right of the menu bar. This is what Firefox (which has its own built-in method of hiding the title bar) does.

**Comment by ukch** [comment]
Here's a screenshot of Firefox's native behaviour:
![Screenshot](https://cloud.githubusercontent.com/assets/973865/5661848/7e2e1ad2-9739-11e4-858d-ece216875d8d.png)

**Comment by wilfm** [comment]
You should be able to use the keyboard shortcuts to resize windows:
You can use keyboard shortcuts to resize the window:
![](https://camo.githubusercontent.com/f86fb0dc4d05f602e58edf63d42a8d95f41961f0/68747470733a2f2f692e696d6775722e636f6d2f4c794e4c504f372e706e67)

It should also be possible to resize the window by dragging on the panel to move the window.

Firefox interface is customizable -most programs don't allow this and also patching every program would be painful. I did suggest the windows buttons extension at gnome shell extensions, but I don't think this has been updated since 3.8. Which version of gnome she'll are you using?

**Comment by zazy** [comment]
You can also try this: keep the ALT key pressed and click with your mouse anywhere on the maximized window. The cursor will become a "fist grabbing the window". Now move the mouse and the window will restore its previous size.
I prefer this method because it will not force you to take away your hand from the mouse.

**Comment by ukch** [comment]
@zazy that doesn't seem to work for me (although I remember using it in the past). Any idea where that alt-drag behaviour is set?

**Comment by zazy** [comment]
@ukch You cant see which is the modifier for your mouse using dconf-editor (you can launch it from a terminal or from Alt+F2 -> dconf-editor -> (press enter). Expand the tree on the left side so that you can reach the org -> gnome -> desktop -> wm -> preferences schema. On the right side, now, search for the key "mouse-button-modifier". Here you can set the mouse modifier for window click actions. Mine is set to "<Alt>" (without quotes) .

**Comment by a-m-s** [comment]
The default key is not ALT, it's "super", a.k.a. the "windows key".

**Comment by zazy** [comment]
@a-m-s noone told that ALT is the default key. It's distribution dependent. On a new Suse, for example, there is no value set.

**Comment by a-m-s** [comment]
@zazy Super been the default in Gnome since 3.5: https://wiki.gnome.org/Design/OS/KeyboardShortcuts

I'm sorry to hear some distributions have broken the defaults.

**Comment by warrenguy** [comment]
You could use the Window Buttons extension which should do exactly what you suggest: https://extensions.gnome.org/extension/426/window-buttons/

However the version on gnome.org is old and doesn't support 3.14. Here's one that does: https://github.com/danielkza/Gnome-Shell-Window-Buttons-Extension

**Comment by wilfm** [comment]
@warrenguy Wow thanks works here :)
![screenshot from 2015-02-24 21 28 31](https://cloud.githubusercontent.com/assets/6739839/6359570/18852178-bc6c-11e4-9c80-85981fe10126.png)

You can install it by downloading the zip from:
  https://github.com/danielkza/Gnome-Shell-Window-Buttons-Extension/archive/master.zip
Then extract the `window_buttons@biox.github.com` folder in it to `~/.local/share/gnome-shell/extensions/window_buttons@biox.github.com`, restart the shell ((by default <kbd>Alt</kbd>+<kbd>F2</kbd>, enter `r` and press <kbd>enter</kbd>)* and turn on 'Window Buttons' in Gnome Tweak Tool.

I probably should direct people to this in this extensions description - and if @danielkza *(or me (or even integrate some of its functionality back into this one))* could upload the updated extension to https://extensions.gnome.org/ that would be great (probably could also update the README and other stuff, but it works :) - I could also then count this issue as closed.

**Comment by ukch** [comment]
I can enable it, but it doesn't seem to work for me, even after I log back in (I'm on Debian Sid). Any way I can debug the extension to see if there's any error messages?

**Comment by wilfm** [comment]
Works happily on Fedora 21 x64 Gnome 3.14 ... Can't access it at the moment but http://stackoverflow.Com/q/14540012 may help there.

**Comment by danielkza** [comment]
It works for me on Fedora 21 and GNOME 3.14 as @wilfm mentioned. That's where I tested the fixes in. Don't know if there are any others needed, but unless Sid is running a GNOME 3.15 preview it shouldn't be different.

**Comment by wilfm** [comment]
@ukch If you are running gnome 3.15, you may need to edit the second line in [metadata.json](https://github.com/danielkza/Gnome-Shell-Window-Buttons-Extension/blob/master/window_buttons%40biox.github.com/metadata.json#L2) to include 3.15 and restart the shell. You can check your version with `gnome-shell --version`.

**Comment by ukch** [comment]
Nope. Gnome 3.14.2. It doesn't refuse to switch on or anything, it just silently doesn't do anything.
